src/main/scala/org/broadinstitute/hail/variant/VariantSampleMatrix.scala

```diff
@@ -219,6 +219,23 @@ case class VSMSubgen[T](
   tGen: (Variant) => Gen[T]) {
 
   def gen(sc: SparkContext)(implicit tct: ClassTag[T]): Gen[VariantSampleMatrix[T]] =
+    this.distributeSize.flatMap(x => x.genNoResize(sc))
```

Comment:
We have a slight preference for the latter.  I will update the (meager) style guide: https://github.com/broadinstitute/hail/blob/master/style-guide.md.